test/BEST_PRACTICES.md

```diff
@@ -0,0 +1,53 @@
+# Test Best Practices
+
+This is a collection of best practices to write high quality tests. The best practices
+can be referenced in review comments to speed up the review process.
+
+- [Code Style](#code-style)
+- [Test Case Traceability](#test-case-traceability)
+- [Don't Sleep](#dont-sleep)
+- [Independent](#independent)
+- [Secrets](#secrets)
+- [Logging](#logging)
+
+## Code Style
+
+The code style should be consistent with the style used in the integreatly-operator codebase [TODO: link to golang practices used by engineering]
+
+Also, in case of doubt, the Golang standard best practices can be used:
+
+- [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments)
+- [Effective Go](https://golang.org/doc/effective_go.html) documents.
+
+## Test Case Traceability
+
+In order to compute test coverage and test automation progress across multiple test suites all
+automated tests must have an ID and must be traced back to the [integreatly-test-cases](https://gitlab.cee.redhat.com/integreatly-qe/integreatly-test-cases)
+repo following the [How to automate a test case and link it back](https://gitlab.cee.redhat.com/integreatly-qe/integreatly-test-cases#how-to-automate-a-test-case-and-link-it-back)
+tutorial.
+
+## Don't Sleep
+
+When waiting for a resource to become available or an operation to complete do not use fixed wait times and/or sleep statements,
```

Comment:
Very samll point - remove the NEWLINE.   ;-)

Reply:
I removed the newlines from all paragraphs, is that right